e2e/go.mod

```diff
@@ -0,0 +1,279 @@
+module github.com/celestiaorg/celestia-node/e2e
+
+go 1.23.6
```

Comment:
I explicitly chose this version due to a breakage if bumped to v1.24

`link: github.com/bytedance/sonic/loader: invalid reference to runtime.lastmoduledatap`

It looks like the sonic library is pulled in by the cosmos sdk and depends on some go internals that changed between these versions.